### ISSUE TITLE
SyntaxError when running the bot

### ISSUE THREAD

**Issue by mikelei8291** [blueset/ehForwarderBot#1]
When I tried to run the bot, I got the error below:
```
Traceback (most recent call last):
  File "main.py", line 107, in <module>
    init()
  File "main.py", line 60, in init
    obj = getattr(__import__(i[0], fromlist=i[1]), i[1])
  File "/root/github/ehForwarderBot/plugins/eh_wechat_slave.py", line 413
    txt = txt.format({**msg['Text'], **msg['Text']['userInfo']})
                       ^
SyntaxError: invalid syntax
```

Then I just deleted the `**` before the `msg` in the inner dict and the bot ran properly.

---
Also, it seems that photos and files cannot be sent at this moment, not sure if it is caused by the editing of the code.

**Comment by mikelei8291** [comment]
Updated: Photos can be sent and received if chatting with others but not with myself.

**Comment by blueset** [comment]
Issue 1:
Fixed in 1093a6e.

Issue 2:
Not reproduced in local environment. I'll keep this open, if you still can reproduce the bug with the latest `dev` commit, please try to upload the log here.

```
git checkout dev
python3 main.py -l EFB.log -vv
# ^C
cat EFB.log
```

**Comment by mikelei8291** [comment]
Issue 1: fixed by upgrading Python version form 3.4.6 to 3.5.2

---
Issue 2:
log cat:
```
 Processing Update: {'update_id': 161106361, 'message': {'new_chat_photo': [], 'date': 1482473045, 'migrate_to_chat_id': 0, 'migrate_from_chat_id': 0, 'delete_chat_photo': False, 'caption': '', 'group_chat_created': False, 'photo': [{'file_size': 1215, 'height': 60, 'file_id': 'AgADBQADtcIxG4WvdQaEG219aYqctZMsyjIABEDcQbq3WOWosUQBAAEC', 'width': 90}, {'file_size': 17407, 'height': 213, 'file_id': 'AgADBQADtcIxG4WvdQaEG219aYqctZMsyjIABD_cgU2c1ol9skQBAAEC', 'width': 320}, {'file_size': 81210, 'height': 533, 'file_id': 'AgADBQADtcIxG4WvdQaEG219aYqctZMsyjIABFvO99kUSZw_s0QBAAEC', 'width': 800}, {'file_size': 129919, 'height': 853, 'file_id': 'AgADBQADtcIxG4WvdQaEG219aYqctZMsyjIABD_BFhZ4mlohsEQBAAEC', 'width': 1280}], 'supergroup_chat_created': False, 'chat': {'all_members_are_admins': False, 'title': '', 'username': 'MikeLei', 'last_name': 'Lei', 'first_name': 'Mike', 'type': 'private', 'id': 000000000}, 'message_id': 178, 'channel_chat_created': False, 'entities': [], 'from': {'type': '', 'username': 'MikeLei', 'first_name': 'Mike', 'last_name': 'Lei', 'id': 000000000}, 'reply_to_message': {'new_chat_photo': [], 'date': 1482472992, 'migrate_to_chat_id': 0, 'migrate_from_chat_id': 0, 'delete_chat_photo': False, 'caption': '', 'new_chat_title': '', 'group_chat_created': False, 'photo': [], 'supergroup_chat_created': False, 'chat': {'all_members_are_admins': False, 'title': '', 'username': 'MikeLei', 'last_name': 'Lei', 'first_name': 'Mike', 'type': 'private', 'id': 000000000}, 'message_id': 177, 'channel_chat_created': False, 'entities': [], 'from': {'type': '', 'username': 'Bot', 'first_name': "Bot", 'last_name': '', 'id': 000000000}, 'edit_date': 1482473013, 'text': "Reply to this message to chat with 'Mike' from '💬 WeChat Slave'."}, 'new_chat_title': '', 'text': ''}}
2016-12-23 14:04:06,160 - masterTG.plugins.eh_telegram_master - DEBUG
 ----
Msg from tg user:
{'new_chat_photo': [], 'date': 1482473045, 'migrate_to_chat_id': 0, 'migrate_from_chat_id': 0, 'delete_chat_photo': False, 'caption': '', 'group_chat_created': False, 'photo': [{'file_size': 1215, 'height': 60, 'file_id': 'AgADBQADtcIxG4WvdQaEG219aYqctZMsyjIABEDcQbq3WOWosUQBAAEC', 'width': 90}, {'file_size': 17407, 'height': 213, 'file_id': 'AgADBQADtcIxG4WvdQaEG219aYqctZMsyjIABD_cgU2c1ol9skQBAAEC', 'width': 320}, {'file_size': 81210, 'height': 533, 'file_id': 'AgADBQADtcIxG4WvdQaEG219aYqctZMsyjIABFvO99kUSZw_s0QBAAEC', 'width': 800}, {'file_size': 129919, 'height': 853, 'file_id': 'AgADBQADtcIxG4WvdQaEG219aYqctZMsyjIABD_BFhZ4mlohsEQBAAEC', 'width': 1280}], 'supergroup_chat_created': False, 'chat': {'all_members_are_admins': False, 'title': '', 'username': 'MikeLei', 'last_name': 'Lei', 'first_name': 'Mike', 'type': 'private', 'id': 000000000}, 'message_id': 178, 'channel_chat_created': False, 'entities': [], 'from': {'type': '', 'username': 'MikeLei', 'first_name': 'Mike', 'last_name': 'Lei', 'id': 000000000}, 'reply_to_message': {'new_chat_photo': [], 'date': 1482472992, 'migrate_to_chat_id': 0, 'migrate_from_chat_id': 0, 'delete_chat_photo': False, 'caption': '', 'new_chat_title': '', 'group_chat_created': False, 'photo': [], 'supergroup_chat_created': False, 'chat': {'all_members_are_admins': False, 'title': '', 'username': 'MikeLei', 'last_name': 'Lei', 'first_name': 'Mike', 'type': 'private', 'id': 000000000}, 'message_id': 177, 'channel_chat_created': False, 'entities': [], 'from': {'type': '', 'username': 'Bot', 'first_name': "Bot", 'last_name': '', 'id': 000000000}, 'edit_date': 1482473013, 'text': "Reply to this message to chat with 'Mike' from '💬 WeChat Slave'."}, 'new_chat_title': '', 'text': ''}
2016-12-23 14:04:06,160 - masterTG.db.plugins.eh_telegram_master.db - INFO
 get_msg_log 000000000.177
2016-12-23 14:04:06,161 - peewee - DEBUG
 ('SELECT "t1"."master_msg_id", "t1"."text", "t1"."slave_origin_uid", "t1"."slave_origin_display_name", "t1"."slave_member_uid", "t1"."slave_member_display_name", "t1"."msg_type", "t1"."sent_to", "t1"."time" FROM "msglog" AS t1 WHERE ("t1"."master_msg_id" = ?) ORDER BY "t1"."time" DESC', ['000000000.177'])
2016-12-23 14:04:06,161 - masterTG.plugins.eh_telegram_master - DEBUG
 tg: msg type: Photo
2016-12-23 14:04:06,161 - telegram.bot - DEBUG
 Entering: getFile
2016-12-23 14:04:07,741 - telegram.bot - DEBUG
 {'file_size': 129919, 'file_path': 'https://api.telegram.org/file/bot000000000:ohWDxoegtMHetaanRykA0OxSVOENplz1AHL/photo/file_39.jpg', 'file_id': 'AgADBQADtcIxG4WvdQaEG219aYqctZMsyjIABD_BFhZ4mlohsEQBAAEC'}
2016-12-23 14:04:07,741 - telegram.bot - DEBUG
 Exiting: getFile
2016-12-23 14:04:08,770 - SlaveWC.plugins.eh_wechat_slave - INFO
 msg.text
2016-12-23 14:04:08,783 - requests.packages.urllib3.connectionpool - INFO
 Resetting dropped connection: wx.qq.com
2016-12-23 14:04:09,208 - requests.packages.urllib3.connectionpool - DEBUG
 "POST /cgi-bin/mmwebwx-bin/webwxbatchgetcontact?type=ex&r=1482473048 HTTP/1.1" 200 884
2016-12-23 14:04:09,824 - requests.packages.urllib3.connectionpool - DEBUG
 "GET /cgi-bin/mmwebwx-bin/webwxgetcontact?r=1482473049&seq=0&skey=@crypt_4b0fa9a4_c376da800738cf9ffccabb173aaa84e0 HTTP/1.1" 200 14112
2016-12-23 14:04:10,530 - requests.packages.urllib3.connectionpool - DEBUG
 "GET /cgi-bin/mmwebwx-bin/webwxgetcontact?r=1482473049&seq=0&skey=@crypt_4b0fa9a4_c376da800738cf9ffccabb173aaa84e0 HTTP/1.1" 200 14112
2016-12-23 14:04:11,237 - requests.packages.urllib3.connectionpool - DEBUG
 "GET /cgi-bin/mmwebwx-bin/webwxgetcontact?r=1482473050&seq=0&skey=@crypt_4b0fa9a4_c376da800738cf9ffccabb173aaa84e0 HTTP/1.1" 200 14112
2016-12-23 14:04:11,384 - requests.packages.urllib3.connectionpool - DEBUG
 "POST /cgi-bin/mmwebwx-bin/webwxbatchgetcontact?type=ex&r=1482473051 HTTP/1.1" 200 884
2016-12-23 14:04:11,385 - SlaveWC.plugins.eh_wechat_slave - INFO
 Sending message to Wechat:
Target-------
uid: wxid_3otynddtc8vc21
UserName: False
NickName:
2016-12-23 14:04:11,385 - SlaveWC.plugins.eh_wechat_slave - INFO
 Got message of type Image
2016-12-23 14:04:11,385 - SlaveWC.plugins.eh_wechat_slave - INFO
 Image/Sticker Image
2016-12-23 14:04:11,386 - SlaveWC.plugins.eh_wechat_slave - INFO
 Sending Image...
2016-12-23 14:04:11,389 - requests.packages.urllib3.connectionpool - INFO
 Resetting dropped connection: file.wx.qq.com
2016-12-23 14:04:12,487 - requests.packages.urllib3.connectionpool - DEBUG
 "POST /cgi-bin/mmwebwx-bin/webwxuploadmedia?f=json HTTP/1.1" 200 789
2016-12-23 14:04:12,624 - requests.packages.urllib3.connectionpool - DEBUG
 "POST /cgi-bin/mmwebwx-bin/webwxsendmsgimg?fun=async&f=json HTTP/1.1" 200 77
2016-12-23 14:04:12,626 - SlaveWC.plugins.eh_wechat_slave - INFO
 Image sent with result {'LocalID': '', 'BaseResponse': {'ErrMsg': '', 'RawMsg': '', 'Ret': -1}, 'MsgID': ''}
```

IDs are replaced by 000000000, and bot username is edited.

**Comment by blueset** [comment]
Issue 2 seems to be due to the limit of WeChat Web protocol.
